src/models/generated/MultiStartCost.jl

```diff
@@ -0,0 +1,91 @@
+#=
+This file is auto-generated. Do not edit.
+=#
+"""
+    mutable struct MultiStartCost <: OperationalCost
+        variable::VariableCost
+        no_load::Float64
+        fixed::Float64
+        startup::NamedTuple{(:hot, :warm, :cold), NTuple{3, Float64}}
+        shutdn::Float64
+        forecasts::InfrastructureSystems.Forecasts
+        internal::InfrastructureSystemsInternal
+    end
+
+Data Structure Operational Cost Data in Three parts fixed, variable cost and start - stop costs.
+
+# Arguments
+- `variable::VariableCost`: variable cost
+- `no_load::Float64`: no load cost
+- `fixed::Float64`: fixed cost
+- `startup::NamedTuple{(:hot, :warm, :cold), NTuple{3, Float64}}`: startup cost
+- `shutdn::Float64`: shutdown cost, validation range: (0, nothing), action if invalid: warn
+- `forecasts::InfrastructureSystems.Forecasts`: internal forecast storage
+- `internal::InfrastructureSystemsInternal`: power system internal reference, do not modify
+"""
+mutable struct MultiStartCost <: OperationalCost
+    "variable cost"
+    variable::VariableCost
+    "no load cost"
+    no_load::Float64
+    "fixed cost"
+    fixed::Float64
+    "startup cost"
+    startup::NamedTuple{(:hot, :warm, :cold), NTuple{3, Float64}}
+    "shutdown cost"
+    shutdn::Float64
+    "internal forecast storage"
+    forecasts::InfrastructureSystems.Forecasts
+    "power system internal reference, do not modify"
+    internal::InfrastructureSystemsInternal
+end
+
+function MultiStartCost(variable, no_load, fixed, startup, shutdn, forecasts=InfrastructureSystems.Forecasts(), )
+    MultiStartCost(variable, no_load, fixed, startup, shutdn, forecasts, InfrastructureSystemsInternal(), )
+end
+
+function MultiStartCost(; variable, no_load, fixed, startup, shutdn, forecasts=InfrastructureSystems.Forecasts(), )
+    MultiStartCost(variable, no_load, fixed, startup, shutdn, forecasts, )
+end
+
+# Constructor for demo purposes; non-functional.
+function MultiStartCost(::Nothing)
+    MultiStartCost(;
+        variable=VariableCost((0.0, 0.0)),
+        no_load=0.0,
+        fixed=0.0,
+        startup=(hot = Inf, warm = Inf,cold = Inf),
```

Comment:
Inf as default?

Reply:
I don't think anyone has fixed the issue with JSON-serializing `Inf` and `NaN`.

Reply:
I didn't want it to be zero but a large number, as there are cases where a device only has 1 or 2 types of starts.

Reply:
Thing I'll do is add a const `START_COST` and use it instead, which would be a reasonably large number

Reply:
we should then add an individual field for each and make a constant for "high_cost" I am not sure to want Inf to be the value since it might not be interpretable by the solvers.